### PR TITLE
[Android] Add custom system-wide text selection toolbar buttons for SelectableRegion

### PR DESCRIPTION
**Pull request by bleroux**
## Description

This PR adds custom system-wide text selection toolbar buttons on Android for `SelectableRegion` and `SelectionArea`.

https://github.com/flutter/flutter/pull/139738 adds those buttons for `EditableText` (which is used by `TextField` and `SelectableText` but not by `SelectionArea`).

## Related Issue

Step 5 for https://github.com/flutter/flutter/issues/139361

## Tests

Adds 2 tests.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/selectable_region.dart [hunk 1]
```diff
@@ -1203,7 +1217,29 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
...
+        onPressed: () async {
+          final String selectedText = data.plainText;
+          if (selectedText.isNotEmpty) {
+            await _processTextService.processTextAction(action.id, selectedText, true);
+            hideToolbar();
+          }
+        },
+      ));
+    }
+    return buttonItems;
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/selectable_region.dart:1242]: I want to point out that a lot of the code changes in this file are duplicating code that's now in EditableText, too, though I think that's inevitable for now. Hopefully we'll be able to deduplicate this as part of the larger project to unify the text selection code.

**Review by justinmc — APPROVED** [review]
LGTM 👍 , thanks for adding this for non-editable text too.

And by the way, I was out on leave for most of the time that you were working on this, but thanks so much for tackling this overall project. It's such a huge improvement for Flutter on Android.

Edit: I think the test failures may be a flake and you need to merge in master?

File: packages/flutter/lib/src/widgets/selectable_region.dart (hunk 1 above)
Comment (bleroux) [reply packages/flutter/lib/src/widgets/selectable_region.dart:1242]: > Hopefully we'll be able to deduplicate this as part of the larger project to unify the text selection code.

Very interesting, is there an issue to track this project?

Comment (justinmc) [reply packages/flutter/lib/src/widgets/selectable_region.dart:1242]: See https://github.com/flutter/flutter/issues/104547 and https://github.com/flutter/flutter/issues/129583, CC @Renzo-Olivares.

**Pull request merged**